aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -55,15 +55,21 @@ SparseTensor& mul_out_sparse_zerodim(SparseTensor& r, const SparseTensor& t, con
   AT_ASSERT(t.is_sparse());
   AT_ASSERT(value.dim() == 0);
 
+  // Resolve a possibly sparse COO value to a strided tensor.
+  auto& value_ = (value.is_sparse() ? value._values() : value);
```

Comment:
This raises error

```
error C2440: 'initializing': cannot convert from 'at::Tensor' to 'at::Tensor &'
```

_values() doesn't return a reference. 

As a side note, I personally avoid auto if the resulting type isn't super long otherwise, simply because I can avoid unnecessary type conversions etc., which can be costly. But I also understand if that's not practical or if I'm being overly cautious.

Reply:
Fixed. No problem here to avoid auto.

Also enabled a test of multiplying two 0D sparse tensors that hinted that we should also deal with empty sparse tensors here.